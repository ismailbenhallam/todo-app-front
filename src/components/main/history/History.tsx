import { FC, MouseEvent, useState } from "react";
import { TodoState } from "../../../models/Todo";
import CompleteTodo from "../home/Todo/CompleteTodo";
import TodoList from "../home/TodoList";
import { HistoryContainer, Tab, Tabs } from "./History.style";

interface TabsType {
  [key: string]: {
    name: string;
    filterFunctionState: TodoState;
  };
}

const TABS: TabsType = {
  done: {
    name: "Terminées",
    filterFunctionState: TodoState.DONE,
  },
  deleted: {
    name: "Supprimées",
    filterFunctionState: TodoState.DELETED,
  },
};

const History: FC = () => {
  const [activeButton, setActiveButton] = useState("done");

  const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
    setActiveButton(event.currentTarget.name);
  };

  return (
    <HistoryContainer>
      <Tabs>
        {Object.entries(TABS).map((entry) => (
          <Tab
            key={entry[0]}
            onClick={handleClick}
            name={entry[0]}
            className={activeButton === entry[0] ? "active" : ""}>
            {entry[1].name}
          </Tab>
        ))}
      </Tabs>
      {/* as={TodoListContainer} */}
      <TodoList
        todoComponent={CompleteTodo}
        filterFunction={(todo) =>
          todo.state === TABS[activeButton].filterFunctionState
        }
      />
    </HistoryContainer>
  );
};

export default History;
