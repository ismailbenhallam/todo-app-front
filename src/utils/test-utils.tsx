import { render, RenderOptions } from "@testing-library/react";
import { FC, ReactElement } from "react";
import { Provider } from "react-redux";
import store from "redux/store";

export const AllProviders: FC = ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
