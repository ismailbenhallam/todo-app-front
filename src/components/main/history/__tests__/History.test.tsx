import renderer from "react-test-renderer";
import History from "..";
import { AllProviders } from "../../../../utils/test-utils";

test("History should match snapshot", () => {
  const tree = renderer
    .create(
      <AllProviders>
        <History />
      </AllProviders>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
