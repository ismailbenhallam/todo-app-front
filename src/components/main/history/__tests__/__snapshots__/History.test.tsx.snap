// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History should match snapshot 1`] = `
<div
  className="sc-bqyKOL GsNPv"
>
  <div
    className="sc-crrszt gmEEbj"
  >
    <button
      className="sc-dQoVA hfYaBt active"
      name="done"
      onClick={[Function]}
    >
      Terminées
    </button>
    <button
      className="sc-dQoVA hfYaBt"
      name="deleted"
      onClick={[Function]}
    >
      Supprimées
    </button>
  </div>
  <div
    className="sc-kEjbQP bOzHSf"
    data-testid="TodoListContainer"
  >
    No data
  </div>
</div>
`;
